processor/transformprocessor/README.md

```diff
@@ -726,6 +726,9 @@ transform:
 
 The configuration can be used also with `delete_matching_keys()` to copy the attributes that do not match the regular expression.
 
+## Additional OTTL Functions
+The Factory Options `WithAdditionalMetricFunctions`, `WithAdditionalSpanFunctions` and `WithAdditionalLogFunctions` can be set in the transform processor `NewFactory` to add additional OTTL functions to the resulting processor. The `Config` methods `ValidateMetricStatements`, `ValidateSpanStatements` and `ValidateLogStatements` to receive the additional OTTL functions to validate the context statements. By setting a custom `NewFactory` and a custom `Config`, the user can create custom `transform` processor with any additional OTTL functions.
```

Comment:
`s/set in/set when calling/`

I don't think there's any need to document the change to the `Validate*` methods in the README.

What is the "custom `Config`" you're referring to?

Reply:
> What is the "custom Config" you're referring to?

My initial approach consisted in having to define a `NewFactory` with the additional functions and a new `Config.Validate` method using `ValidateWithAdditionalFunctions` to add this functions to the `Validate()` methods.

Reply:
I wouldn't add any API docs here actually, the README is more like an user-facing docs, and normally does not contains API docs. We could document it at the code level instead, using the standard godocs comments.

Reply:
Done! I removed the `README` update and i'll make sure the `FactoryOption` and `NewFactory` have godoc comments.